src/parser/ast.rs

```diff
@@ -44,11 +67,85 @@ pub struct BinModifier {
     /// If they are not this field is None.
     pub card: VectorMatchCardinality,
     /// on/ignoring on labels
-    pub matching: VectorMatchModifier,
+    pub matching: Option<VectorMatchModifier>,
     /// If a comparison operator, return 0/1 rather than filtering.
     pub return_bool: bool,
 }
 
+impl Default for BinModifier {
+    fn default() -> Self {
+        Self {
+            card: VectorMatchCardinality::OneToOne,
+            matching: None,
+            return_bool: false,
+        }
+    }
+}
+
+impl BinModifier {
+    pub fn default_modifier() -> Self {
```

Comment:
Is this method necessary? Looks like we can just use `BinModifier::default()`, or rename this to `new`
```suggestion
    pub fn new() -> Self {
```